### PR TITLE
Add vendor key support

### PR DESCRIPTION
**Pull request by michaelthomasj**

### CODE REVIEW

File: include/psa/crypto_extra.h [hunk 1]
```diff
@@ -66,6 +66,70 @@ extern "C" {
...
+ * \brief Generate symmetric key of vendor defined format.
+ *
+ * \warning This function **can** fail! Callers MUST check the return status
+ *          and MUST NOT use the content of the output buffer if the return
+ *          status is not #PSA_SUCCESS.
+ *
+ * \note    This function has to be defined by the vendor if MBEDTLS_PSA_CRYPTO_ACCEL_DRV_C 
+ *          is defined.
+ *          A weakly linked version is provided by default and returns
+ *          PSA_ERROR_NOT_SUPPORTED. Do not use this function directly;
```
[the remark this answers is not in the document]
Comment (michaelthomasj) [reply include/psa/crypto_extra.h]: Done.

File: include/psa/crypto_values.h [hunk 2]
```diff
@@ -1501,6 +1504,18 @@
  */
 #define PSA_KEY_LIFETIME_PERSISTENT             ((psa_key_lifetime_t)0x00000001)
 
+#define PSA_KEY_LIFETIME_IS_VOLATILE(lifetime) \
```
Comment (gilles-peskine-arm) [inline include/psa/crypto_values.h]: Please refer to the definition of `PSA_KEY_LIFETIME_IS_VOLATILE` in https://github.com/ARMmbed/mbed-crypto/pull/358, which is slated to be included in the next release of Mbed Crypto and in an upcoming version of the PSA Cryptography specification.

File: include/psa/crypto_values.h [hunk 3]
```diff
@@ -380,6 +380,9 @@
  */
 #define PSA_KEY_TYPE_AES                        ((psa_key_type_t)0x40000001)
 
+/** Whether a key type is AES. */
+#define PSA_KEY_TYPE_IS_AES(type) (((type)&PSA_KEY_TYPE_AES) != 0)
```
[the remark this answers is not in the document]
Comment (michaelthomasj) [reply include/psa/crypto_values.h]: Agreed.

File: include/psa/crypto_values.h (hunk 2 above)
Comment (michaelthomasj) [reply include/psa/crypto_values.h]: I am fine with that new definition. Im not quite sure how to get it in here though.

File: include/psa/crypto_values.h [hunk 4]
```diff
@@ -1501,6 +1504,18 @@
...
+
+#define PSA_KEY_LIFETIME_IS_PERSISTENT(lifetime) \
+    (((lifetime)&PSA_KEY_LIFETIME_PERSISTENT) != 0)
+
+#define PSA_KEY_LIFETIME_VENDOR_FLAG ((psa_key_lifetime_t)0x80000000)
+#define PSA_KEY_LIFETIME_PERSISTENT_VENDOR (PSA_KEY_LIFETIME_VENDOR_FLAG | PSA_KEY_LIFETIME_PERSISTENT)
+#define PSA_KEY_LIFETIME_VOLATILE_VENDOR (PSA_KEY_LIFETIME_VENDOR_FLAG | PSA_KEY_LIFETIME_VOLATILE)
+
+#define PSA_KEY_LIFETIME_IS_VENDOR_DEFINED(lifetime) \
+    (((lifetime)&PSA_KEY_LIFETIME_VENDOR_FLAG) != 0)
```
[the remark this answers is not in the document]
Comment (michaelthomasj) [reply include/psa/crypto_values.h]: This one will also change to match your proposal.

File: include/psa/crypto_struct.h [hunk 5]
```diff
@@ -378,7 +378,7 @@ static inline void psa_set_key_lifetime(psa_key_attributes_t *attributes,
                                         psa_key_lifetime_t lifetime)
 {
     attributes->core.lifetime = lifetime;
-    if( lifetime == PSA_KEY_LIFETIME_VOLATILE )
+    if (PSA_KEY_LIFETIME_IS_VOLATILE(lifetime))
```
[the remark this answers is not in the document]
Comment (michaelthomasj) [reply include/psa/crypto_struct.h]: Fixed

**Pull request merged**